xbmc/filesystem/FileDirectoryFactory.cpp

```diff
@@ -64,20 +64,22 @@ IFileDirectory* CFileDirectoryFactory::Create(const CURL& url, CFileItem* pItem,
 
   std::string strExtension=URIUtils::GetExtension(url);
   StringUtils::ToLower(strExtension);
-  VECADDONS codecs;
-  CAddonMgr::GetInstance().GetAddons(codecs, ADDON_AUDIODECODER);
-  for (size_t i=0;i<codecs.size();++i)
+  if (!strExtension.empty())
```

Comment:
I don't like that GetAddons is called for every one and single directory and file that goes through our factory ... even if it only makes sense for files with an extensions. I don't see it as workaround, it's an optimization.